gbn/config.go

```diff
@@ -0,0 +1,62 @@
+package gbn
+
+import "time"
+
+// config holds the configuration values for an instance of GoBackNConn.
+type config struct {
```

Comment:
:fire::rocket:! Just curious how this should be implemented along with the `TimeoutManager` from #88. Should the config be seen as the initial configuration by the user, but then the values in the `TimeoutManager` should be the "actual" values the gbn packages use, i.e. the source of truth? Or should the `TimeoutManager` actively change the values of the config, and the configuration will hold the "source of truth"? I'd assume the prior.

Reply:
i think for the TimeoutManager PR you can just change this struct to hold a list of functional options used to adjust the instantiation of the timeout manager. Does that make sense? can chat about it offline too :)

Reply:
I think I understand what you mean, but will chat with you offline if it's unclear when I implement this :)